lib/streamlit/elements/markdown.py

```diff
@@ -13,15 +13,22 @@
 # limitations under the License.
 
 from typing import cast
+from typing import Optional
+from typing import TYPE_CHECKING
+from typing import Union
 
-import streamlit
 from streamlit import type_util
 from streamlit.proto.Markdown_pb2 import Markdown as MarkdownProto
 from .utils import clean_text
 
+if TYPE_CHECKING:
+    import sympy
```

Comment:
Are there any implications of importing this here when the `sympy` package isn't a required dependency?

I'm guessing this all works correctly as the typechecking stage of CI passes even without adding the package to requirements as far as I can tell, but I'm not entirely sure how/why

Reply:
My understanding on this isn't perfect, but I think `mypy` just interprets non-installed packages as the `Any` type, hence what I'm doing here will always work as long as `sympy` isn't installed. But technically, if someone later on decides on having `sympy` being installed at type check time, it might uncover new type errors that are currently hidden by the `Any` treatment. That's a minor issue, though; it just might be confusing to whoever adds the dependency.

For more info: https://mypy.readthedocs.io/en/stable/running_mypy.html#missing-imports